aten/src/ATen/core/dispatch/OperatorEntry.h

```diff
@@ -71,8 +76,11 @@ class OperatorEntry final {
   void prepareForDeregistration();
 
   // Postcondition: caller is responsible for disposing of the kernel
-  std::list<KernelSchemaPair>::iterator registerKernel(c10::optional<DispatchKey> dispatch_key, KernelFunction kernel, std::unique_ptr<FunctionSchema> inferred_function_schema);
-  void deregisterKernel_(c10::optional<DispatchKey> dispatch_key, std::list<KernelSchemaPair>::iterator kernel);
+  std::list<KernelEntry>::iterator registerKernel(c10::optional<DispatchKey> dispatch_key, KernelFunction kernel, std::unique_ptr<FunctionSchema> inferred_function_schema, std::string debug);
+  void deregisterKernel_(c10::optional<DispatchKey> dispatch_key, std::list<KernelEntry>::iterator kernel);
+
+  std::string dumpState() const;
+  void checkInvariants() const;
```

Comment:
Yeah, I vacillated a little about this. In the end, I added one test to `test/test_dispatch.py` to check if invariants were satisfied for all operators that were registered. There isn't a convenient place to call this function in prod; if you call it at registration time, you will keep testing invariants for every registration, and I imagine it will be noticeable at that point. You definitely want to be able to turn off invariant checking, because in general it might be expensive to do this check.